src/lib.rs

```diff
@@ -39,25 +47,38 @@ pub struct SphinxPacket {
 
 impl SphinxPacket {
     pub fn new(
-        message: Vec<u8>,
+        mut message: Vec<u8>,
         route: &[Node],
         destination: &Destination,
         delays: &[Delay],
-    ) -> Result<SphinxPacket, SphinxUnwrapError> {
+        surb_material: Option<SURBMaterial>,
+    ) -> Result<SphinxPacket, SphinxError> {
         let initial_secret = crypto::generate_secret();
         let (header, payload_keys) =
             header::SphinxHeader::new(initial_secret, route, delays, destination);
 
+        if let Some(surb_material) = surb_material {
+            let surb_initial_secret = crypto::generate_secret();
+            let surb = SURB::new(
+                surb_initial_secret,
+                &surb_material.surb_route,
+                &surb_material.surb_delays,
+                &surb_material.surb_destination,
+            )
+            .unwrap();
```

Comment:
Since `SURB::new()` returns possibly same error type as `SphinxPacket::new()`, don't unwrap the result (which may cause panic if we change something in `SURB::new()` and forget about it) but rather handle it with a `?` ?